modules/common/memcached/memcached.go

```diff
@@ -0,0 +1,44 @@
+package memcached
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	memcachedv1 "github.com/openstack-k8s-operators/infra-operator/apis/memcached/v1beta1"
+	oko_svc "github.com/openstack-k8s-operators/lib-common/modules/common/service"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s_labels "k8s.io/apimachinery/pkg/labels"
+)
+
+// NewMemcached function will return a memcached v1beta1 instance of the CR. This can then be used
+// with controllerutil.CreateOrPatch() to create the resource in the cluster.
+func NewMemcached(name string, namespace string) *memcachedv1.Memcached {
+	return &memcachedv1.Memcached{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "memcached.openstack.org/v1beta1",
+			Kind: "Memcached",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Namespace: namespace,
+		},
+	}
+}
+
+// GetMemcachedServiceName function searches for all memcached services matching the provided memcachedName
+func GetMemcachedServiceName(ctx context.Context, memcachedName string, h *helper.Helper, namespace string) (svc string, err error) {
+	labelSelector := k8s_labels.Set{"app": "memcached"}
+	svcList, err := oko_svc.GetServicesListWithLabel(ctx, h, namespace, labelSelector)
+
+	if err != nil {
+		return "", err
+	}
+	for _, service := range svcList.Items {
+		if service.Name == memcachedName {
```

Comment:
See above, this can't be used as is because of  "statefulset.kubernetes.io/pod-name": "memcached-0"

Reply:
resolved from Brendan's previous comment. This was not the intent of this function.